game/world/objects/npc.cpp

```diff
@@ -471,8 +472,15 @@ bool Npc::resetPositionToTA() {
   setDirection(at->dirX,at->dirY,at->dirZ);
   owner.script().fixNpcPosition(*this,0,0);
 
-  if(!isDead)
+  if(!isDead) {
     attachToPoint(at);
+    if(!isMonster())
+      invent.autoEquipWeapons(*this);
+    }
+
+  if(g2)
+    owner.script().invokeRefreshAtInsert(*this);
```

Comment:
do you need to call this for dead npc's as well?

Reply:
Good Point. Testing shows script is called for dead npcs. But first line directly returns in that case.
```C
func void B_RefreshAtInsert()
{
	// EXIT IF...
	
	// ------ Tote = Exit ------
	if (self.attribute[ATR_HITPOINTS] <= 0)
	{
		return;
	};
```